ui/tract-renderer/src/main/java/org/cru/godtools/tract/widget/PageContentLayout.java

```diff
@@ -406,6 +430,65 @@ protected LayoutParams generateLayoutParams(final ViewGroup.LayoutParams p) {
         return generateDefaultLayoutParams();
     }
 
+    //region Card Bounce Animation
+
+    /**
+     * This method will either create a new bounce Animation or start the one that is already created.
+     */
+    @UiThread
+    private void animateFirstCardView() {
+        if (mActiveCard != null) {
+            return;
+        }
+
+        for (int i = 0; i < getChildCount(); i++) {
+            final View child = getChildAt(i);
+            final LayoutParams lp = (LayoutParams) child.getLayoutParams();
+            switch (lp.childType) {
+                case CHILD_TYPE_CARD:
+                    bounceCardAnimation(child);
+                    return; // Stop after first Card View
+            }
+        }
+        stopBounceAnimation();
+
+    }
+
+    public void setAnimateCard(boolean animate) {
+        this.mBounceFirstCard = animate;
+        if (mBounceFirstCard) {
+            mHandler.sendEmptyMessageDelayed(BOUNCE_MESSAGE, DELAY_FIRST_BOUNCE_ANIMATION);
+        }
+    }
+
+    /**
+     * This method will stop the bounce Animation by setting the AnimateCard variable to false.
+     */
+    private void stopBounceAnimation() {
```

Comment:
I don't think this method is needed anymore. `mBounceFirstCard` is managed externally only.

Reply:
In the animateFirstCardView() I am calling stopBounceAnimation so the loop doesn't continue on a page with no cards.  If you would like to remove the method I can just set the boolean value to false instead of calling the method.

Reply:
hmm, I think there might be an overloaded meaning here. the `mBounceFirstCard` boolean should just indicate if this view should animate the first card when one is available. if you are clearing the flag here the view won't start animating the first card if one is added to the view later.

Reply:
I feel that we should never be changing the value of `mBounceFirstCard` within the `PageContentLayout`, any changes to that value should be initiated externally.

Reply:
want to talk this through?